### PR TITLE
Move Java Google BOM to best effort post-release task

### PR DESCRIPTION
**Pull request by damccorm**
This should ideally be done well in advance of the release, and should be best effort from the release manager

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: contributor-docs/release-guide.md [hunk 1]
```diff
@@ -1356,6 +1346,41 @@ Also, update [the Wikipedia article on Apache Beam](https://en.wikipedia.org/wik
...
 
+#### Update the Java BOM
+
+Google releases a BOM that pins compatible versions of their Java libraries.
+After the release, try updating the BOM to the latest version.
+
+To do so, create a draft PR and run test suites following the instructions at
+https://github.com/apache/beam/blob/master/contributor-docs/java-dependency-upgrades.md.
+
+If there are no test failures, request review and merge the PR as normal.
```
Comment (kennknowles) [inline contributor-docs/release-guide.md:1357]: Given the state of our CI and tests, a little more active triage than just looking at the status.

Comment (damccorm) [reply contributor-docs/release-guide.md]: I had kinda covered it in the next sentence, but I updated the wording to be more explicit

**Comment by tvalentyn** [comment]
I'll defer to @kennknowles who is already reviewing.

**Review by kennknowles — APPROVED** [review]
fab

**Pull request merged**